### DISCUSSION TITLE
Be able to monetize github repos

### DISCUSSION THREAD

**Discussion by hughperkins** [community/community#19378] [Other Feature Feedback, Questions, & Ideas]
It's possible to monetize youtube videos. Youtube gets money from youtube members, and by showing ads. You get money as a video creator, from youtube, by drawing people to watch youtube, and staying on the site.

I feel it could be interesting to have a similar monetization approach for github repos: one gets paid by github for people who view one's repo, and/or clone it.

**Comment by PeterPierinakos** [comment]
```py
# freemoneygenerator.py

import subprocess

def main():
    while True:
        subprocess.run(['curl', 'https://github.com/account/repository'])
        subprocess.run(['git', 'clone', 'https://github.com/account/repository'])   
```

**Comment by virtyaluk** [comment]
for what exactly GitHub should pay you? for looking at your profile?

**Comment by hughperkins** [comment]
github makes money from people using github. How exactly they monetize that is not fixed in stone. Youtube started off as being completely free to use. Later on they introduced ways to monetize their product, i.e. the videos that users upload. github I assume also started off free. Now there are various ways that github monetizes, primarily by charging people for enterprise, and for private repos. I agree that if they wanted to be able to pay people for creating content that attracts viewers, then they would need to themselves tweak their own monetization in order to get money from viewers. That's clearly possible, since Youtube has achieved  that. github might not be the company that succeeds in doing that. Maybe I myself will ultimately create a company that does that. Or maybe it will be some other company.
